sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryContentClient.java

```diff
@@ -236,9 +223,21 @@ public UploadRegistryBlobResult uploadBlob(BinaryData content) {
      * @throws NullPointerException thrown if the {@code stream} is {@code null}.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public UploadRegistryBlobResult uploadBlob(ReadableByteChannel content, Context context) {
+    public UploadRegistryBlobResult uploadBlob(BinaryData content, Context context) {
         Objects.requireNonNull(content, "'content' cannot be null.");
-        return runWithTracing(UPLOAD_BLOB_SPAN_NAME, (span) -> uploadBlobInternal(content, span), context);
+
+        InputStream stream = content.toStream();
+        ReadableByteChannel channel = Channels.newChannel(stream);
```

Comment:
Is converting the BinaryData to a `ReadableByteChannel` the right way to go about this? For now, I think it's fine as this is implementation logic and can be changed in the future.

Reply:
good point, switched to stream